src/plugins/data_views/public/services/has_data.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreStart, HttpStart } from 'kibana/public';
+import { FLEET_ASSETS_TO_IGNORE } from '../../common';
+import { HasDataViewsResponse, IndicesResponse, IndicesResponseModified } from '../';
+
+export class HasData {
+  private removeAliases = (source: IndicesResponseModified): boolean => !source.item.indices;
+
+  private isUserDataIndex = (source: IndicesResponseModified): boolean => {
+    // filter out indices that start with `.`
+    if (source.name.startsWith('.')) return false;
```

Comment:
It is still possible to create indices starting with a dot. I believe this check is unnecessary?

Reply:
Do we have assurance that all system indices are registered as such so we no longer need to rely on the leading `.`? Historically, any index that starts with a period is treated as a system or hidden index whether it was user created or not.

Reply:
Issue created for this - https://github.com/elastic/kibana/issues/127869